chore: remove public piped instances

FineFindus opened the pull request:
Drops support for public Piped instances. This is due to
-  only a few (2) public instances still being online, neither of them supports streaming
-  Piped development has stalled and due to increasing blocking from YouTube it's doubtful the situation will improve in the near future
- the instances API has been offline for months and the fallback list of active instances is unmaintained
- users encounter cryptic errors and failing signups due to this

This does not remove support for piped instances overall, custom instances are still supported, with the assumption that self-hosting users are knowledgeable enough to handle the issues themselves.

Future steps may include to merge the fetch instance with auth instance, as self-hosting users are unlikely to host two instance, and use one just for fetching channels.

Ref: https://github.com/libre-tube/LibreTube/pull/7915
Closes: https://github.com/libre-tube/LibreTube/issues/7833
Closes: https://github.com/libre-tube/LibreTube/issues/8224
Closes: https://github.com/libre-tube/LibreTube/issues/8251

FineFindus committed:
feat: remove support for public piped instances

Drops support for public Piped instances. This is due to only a few (2)
public instances still being online, neither of them supports streaming.
As Piped development has stalled and due to increasing blocking from YouTube
it's doubtful the situation will improve in the near future.

Removing public Piped instances improves the user experience, as there are no
longer cryptic errors displayed, sign-ups fails, etc.

This does not remove support for piped instances overall, custom instances are
still supported, with the assumption that self-hosting users are knowledgeable
enough to handle the issues themselves.

Ref: https://github.com/libre-tube/LibreTube/pull/7915
Closes: https://github.com/libre-tube/LibreTube/issues/7833

FineFindus committed:
feat: default to local extraction

FineFindus committed:
feat: remove WelcomeActivity

Removes the WelcomeActivity, as it no longer servers a purpose, with the
removal of public Piped instances. For most users this screen only entails
tapping on OK, whilst a minority may use the restore option, which is also
accessible in the preferences.

FineFindus committed:
feat: add 'Add Instance' button to instance dialog

Bnyro reviewed (commented):
> only a few (2) public instances still being online, neither of them supports streaming
>
The thing is that these two still allow you to search and load other data more anonymously - so using the "local streams extraction" with a public Piped instance is still a valid use case imo.

ULTRA-LEGENDS commented:
But it may stop working soon or in the future and we do not wait until it stops or until we receive new issue reports to start making fixes again

Figim commented:
We can continue this work in the future by using a site like Peertube. Peertube needs a user-friendly interface like LibreTube. There is also a concept called Peertube instances. This does not solve the YT problem but also simply changes the purpose and content of LibreTube. So there is no need for an alternative frontend title just for YT.  "An interface created for your favorite stream platform". You mention in the readme that you are using Extractor.

FineFindus committed:
feat(Preference): remove Piped Proxy setting

As the streaming URLs are Ip-specific and signed (thus we can't simply update
the assigned IP) and the proxy/server has a different IP, it's no longer
possible to directly load content from the client for a server assigned URL.

Bnyro commented:
> - do nothing, everything will still work, but they will have to manually add the piped instance if they ever want to change to it again
> - migrate them to full local mode, even if the currently have a working instance
> - migrate the selected fetch instance to a custom instance (we're currently only storing the API URL, but not the Name or the frontend URL).
>
We probably shouldn't yet change the settings for users that have a Piped instance configured. I think there still are some people that use Piped for everything (except video streaming, i.e. `Local stream extraction` enabled) or for authentication / syncing. So I think it's the best to just not do any migration and keep their preferences as they were.
